account_cancel_with_reversal/models/account_voucher.py

```diff
@@ -0,0 +1,37 @@
+# -*- coding: utf-8 -*-
+
+from openerp import models, fields, api
+
+
+class AccountVoucher(models.Model):
+    _inherit = "account.voucher"
+
+    cancel_move_id = fields.Many2one(
+        'account.move',
+        'Cancelled Journal Entry',
+        copy=False
+        )
+
+    @api.multi
+    def cancel_voucher(self):
+        """ Overwrite """
```

Comment:
Is this "Overwriting" necessary? Better to use  Monkey Patch (as I discuss with you)